pkg/skaffold/kubernetes/context/context.go

```diff
@@ -44,19 +44,15 @@ var (
 // When given, the firstCliValue always takes precedence over the yamlValue.
 // Changing the kube-context of a running Skaffold process is not supported, so
 // after the first call, the kube-context will be locked.
-func ConfigureKubeConfig(cliKubeConfig, cliKubeContext, yamlKubeContext string) {
-	newKubeContext := yamlKubeContext
-	if cliKubeContext != "" {
-		newKubeContext = cliKubeContext
-	}
+func ConfigureKubeConfig(cliKubeConfig, cliKubeContext string) {
 	configureOnce.Do(func() {
```

Comment:
In an ideal world, `pkg/skaffold` should be treated like a library, and these values should be computed in `cmd/skaffold` and provided into the library.

There are also the `KUBECONFIG` environment variable that should influencing this decision (and note that it is a `PATH`-like variable that can specify multiple files).